usecases/wip.md

```diff
@@ -0,0 +1,94 @@
+- User1 start the game with `StartGameEvent`
+    - Server creates a new GameState for the current party.
+    - A `GameStateEvent` is sent to each user containing the GameState for the specific user
+        - (The activeUser object in the `GameStateEvent` is null/undefined. This indicates the start-voting-mechanism)
+
+
+## Player Voting to start
+> Continuuation of "Party creation and joining" with User1 and User2
+
+- Players recieve a `GameStateEvent` with empty activePlayer
+- User1 selects a Player to start
+    - User1 sends `PlayerStartVoteEvent` with the selected nominee
+- User2 selects a Player to start
+    - User2 sends `PlayerStartVoteEvent` with the selected nominee
+- Server recieved a `PlayerStartVoteEvent` from each user
+    - Server sends a `GameStateEvent` where the user with the most votes is the "activePlayer"
+
+# Player turn
+> 
+
+- Player recieves a `GameStateEvent` where the "activePlayer" object is the Player
+- Player selects cards to put on the stacks
+    - Player sends `PlayCardsEvent` with the cards and the corresponding stacks
+- Server receives `PlayCardsEvent`
+    - Server validates the JWT and checks if the player has the turn
+    - Server calculates new GameState
+    - Server sends new `GameStateEvent` with new active player
+- Player recvieves new GameState and waits ...
+
+
+# Player has no cards left
+- Player recieves a `GameStateEvent` where the "activePlayer" object is the Player
+- Player selects cards to put on the stacks
+    - Player sends `PlayCardsEvent` with the cards and the corresponding stacks
+- Server receives `PlayCardsEvent`
+    - Server validates the JWT and checks if the player has the turn
+    - Server calculates new GameState
+        - Server notices empty drawStack and no cards left in player hand
+        - Server send `PlayerWonEvent` to all Players
+    - Server sends new `GameStateEvent`
+
+
+# Player cant play anymore
+- Player recieves a `GameStateEvent` where the "activePlayer" object is the Player
+- Player selects cards to put on the stacks
+    - Player sends `PlayCardsEvent` with the cards and the corresponding stacks
+- Server receives `PlayCardsEvent`
+    - Server validates the JWT and checks if the player has the turn
+    - Server calculates new GameState
+        - Server notices Player cant play any cards
+        - Server send `PlayerLostEvent` to all Players
+    - Server sends new `GameStateEvent`
+
+# Player makes an invalid move
+ Player recieves a `GameStateEvent` where the "activePlayer" object is the Player
+- Player selects cards to put on the stacks
+    - Player sends `PlayCardsEvent` with the cards and the corresponding stacks
+- Server receives `PlayCardsEvent`
+    - Server validates the JWT and checks if the player has the turn
+    - Server notices an invalid move (eg. only one card played while drawStack > 0)
+    - Server sends a `InvalidPlayError` to Player
+    - Server waits for new Play from user ...
```

Comment:
As far as i can tell, this can not break the game in a way that can not be achieved a lot easier with just staying idle in your turn...

The whole issue of timeouts is not addressed in the specification, and might not be relevant in the api-spec since it might be implementation dependent. 

I would leave the exact specifics of how the timeouts work up to the ones implementing, and add a section about timeouts explaining possible "attack vectors". Otherwise, we would need to specify the workflow of the backend somewhere in the specification ...

Do you think this is enough?

Reply:
> As far as i can tell, this can not break the game in a way that can not be achieved a lot easier with just staying idle in your turn...

Yeah, I agree, this should be fine